rosidl_parser/rosidl_parser/definition.py

```diff
@@ -410,46 +530,49 @@ class Message:
 
     __slots__ = ('structure', 'constants')
 
-    def __init__(self, structure):
+    def __init__(self, structure: Structure):
         """
         Constructor.
 
-        :param Structure structure: the structure of the message
+        :param structure: the structure of the message
         """
         super().__init__()
         assert isinstance(structure, Structure)
         self.structure = structure
-        self.constants = OrderedDict()
+        self.constants = []
```

Comment:
Everywhere this is used only `.values()` is being access. Therefore changing the data structure accordingly. The parser already ensures that e.g. no duplicates are being added.